### PR TITLE
Bumped Xcode Version to 15.0

### PR DESCRIPTION
**Pull request by stechiu**
### Summary of changes

- Bumped Xcode version to 15.0 minimum, as per Apple requirements https://developer.apple.com/news/?id=khzvxn8a
- Updated repo CHANGELOG and README files

### Checklist

- [x] Added a changelog entry

- @stechiu

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

File: Braintree.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3386,7 +3386,7 @@
 				HEADER_SEARCH_PATHS = "";
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/BraintreeCore/Info.plist";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 14.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
Comment (jaxdesmarais) [inline Braintree.xcodeproj/project.pbxproj:3389]: I don't think we want to change our iOS deployment target, just the Xcode version right? I didn't initially see anything in the Apple requirements indicating that iOS 14 is no longer supported, but maybe I missed it.

File: Demo/Demo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -733,7 +733,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_TEAM = 43253H4X22;
 				INFOPLIST_FILE = MockVenmo/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 14.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
Comment (jaxdesmarais) [inline Demo/Demo.xcodeproj/project.pbxproj:736]: Same question here on the deployment target changes.

**Comment by stechiu** [comment]
Also bumped Xcode and MacOS versions in the script

File: Braintree.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jaxdesmarais) [reply Braintree.xcodeproj/project.pbxproj]: Lots of versions going around so totally understandable to get lost in the version madness. 😄

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,12 @@
 # Braintree iOS SDK Release Notes
 
+## unreleased
+* Require Xcode 15.0 (per [App Store requirements](https://developer.apple.com/news/?id=khzvxn8a))
```
Comment (jaxdesmarais) [inline CHANGELOG.md:4]: Super nit:
```suggestion
* Require Xcode 15.0+ (per [App Store requirements](https://developer.apple.com/news/?id=khzvxn8a))
```

File: Demo/Demo.xcodeproj/project.pbxproj (hunk 2 above)
Comment (jaxdesmarais) [reply Demo/Demo.xcodeproj/project.pbxproj:736]: I think we also want to revert these demo app deployment targets back to 14. Generally we try to keep our demo app at the same minimum version our SDK supports.

File: README.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ Welcome to Braintree's iOS SDK. This library will help you accept card and alter
 
 v6 is the latest major version of Braintree iOS. To update from v5, see the [v6 migration guide](https://github.com/braintree/braintree_ios/blob/main/V6_MIGRATION.md). If you have not yet migrated to v5, see the [v5 migration guide](https://github.com/braintree/braintree_ios/blob/5.x/V5_MIGRATION.md)
 
-**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 14.3+ and Swift 5.8+.
+**The Braintree iOS SDK permits a deployment target of iOS 15.0 or higher**. It requires Xcode 15.0+ and Swift 5.8+.
```
Comment (jaxdesmarais) [inline README.md:13]:
```suggestion
**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 15.0+ and Swift 5.8+.
```

File: Demo/Demo.xcodeproj/project.pbxproj (hunk 2 above)
Comment (stechiu) [reply Demo/Demo.xcodeproj/project.pbxproj:736]: Strange I thought I reverted then all but didn't. Should be good now!

Comment (jaxdesmarais) [reply Demo/Demo.xcodeproj/project.pbxproj:736]: 🤔 I'm not sure that commit got pushed since we are still seeing a diff here.

Comment (stechiu) [reply Demo/Demo.xcodeproj/project.pbxproj:736]: Oops sorry, should be fixed now

File: README.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ Welcome to Braintree's iOS SDK. This library will help you accept card and alter
 
 v6 is the latest major version of Braintree iOS. To update from v5, see the [v6 migration guide](https://github.com/braintree/braintree_ios/blob/main/V6_MIGRATION.md). If you have not yet migrated to v5, see the [v5 migration guide](https://github.com/braintree/braintree_ios/blob/5.x/V5_MIGRATION.md)
 
-**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 14.3+ and Swift 5.8+.
+**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 15.0+ and Swift 5.8+.
```
Comment (scannillo) [inline README.md:13]: By requiring Xcode 15, aren't we also able to require Swift 5.9+?

**Comment by scannillo** [comment]
Do we have a ticket for doing this on V5 as well?

File: README.md (hunk 5 above)
Comment (stechiu) [reply README.md:13]: Yes, just realized that. I'll make the update

**Review by scannillo — APPROVED** [review]

**Pull request merged**